cargo-spatial/src/download.rs

```diff
@@ -0,0 +1,256 @@
+// Conditionally re-export the correct version of download_cli dependent on platform.
+#[cfg(target_os = "linux")]
+pub use self::linux::*;
+#[cfg(any(target_os = "windows", target_os = "macos"))]
+pub use self::win_osx::*;
+
+use crate::config::Config;
+use crate::opt::DownloadSdk;
+use log::*;
+use std::{
+    fs,
+    path::{Path, PathBuf},
+    process,
+};
+
+enum SpatialWorkerSdkPackage {
+    CApiWin,
+    CApiMac,
+    CApiLinux,
+}
+
+impl SpatialWorkerSdkPackage {
+    fn get_package_name(&self) -> &str {
+        match self {
+            SpatialWorkerSdkPackage::CApiWin => "c-static-x86_64-msvc_mt-win32",
+            SpatialWorkerSdkPackage::CApiMac => "c-static-x86_64-clang_libcpp-macos",
+            SpatialWorkerSdkPackage::CApiLinux => "c-static-x86_64-gcc_libstdcpp_pic-linux",
+        }
+    }
+
+    fn get_relative_target_directory(&self) -> &str {
+        match self {
+            SpatialWorkerSdkPackage::CApiWin => "win",
+            SpatialWorkerSdkPackage::CApiMac => "macos",
+            SpatialWorkerSdkPackage::CApiLinux => "linux",
+        }
+    }
+}
+
+enum SpatialToolsPackage {
+    SchemaCompilerWin,
+    SchemaCompilerMac,
+    SchemaCompilerLinux,
+    SnapshotConverterWin,
+    SnapshotConverterMac,
+    SnapshotConverterLinux,
+}
+
+impl SpatialToolsPackage {
+    fn get_package_name(&self) -> &str {
+        match self {
+            SpatialToolsPackage::SchemaCompilerWin => "schema_compiler-x86_64-win32",
+            SpatialToolsPackage::SchemaCompilerMac => "schema_compiler-x86_64-macos",
+            SpatialToolsPackage::SchemaCompilerLinux => "schema_compiler-x86_64-linux",
+            SpatialToolsPackage::SnapshotConverterWin => "snapshot_converter-x86_64-win32",
+            SpatialToolsPackage::SnapshotConverterMac => "snapshot_converter-x86_64-macos",
+            SpatialToolsPackage::SnapshotConverterLinux => "snapshot_converter-x86_64-linux",
+        }
+    }
+
+    fn get_relative_target_directory(&self) -> &str {
+        match self {
+            SpatialToolsPackage::SchemaCompilerWin
+            | SpatialToolsPackage::SchemaCompilerMac
+            | SpatialToolsPackage::SchemaCompilerLinux => "schema_compiler",
+            SpatialToolsPackage::SnapshotConverterWin
```

Comment:
So I *think* Rust supports a leading `|` character in match branches for situations like this, so you should be able to do:

```
| SpatialToolsPackage::SchemaCompilerWin
| SpatialToolsPackage::SchemaCompilerMac
| SpatialToolsPackage::SchemaCompilerLinux => "schema_compiler",
```

That said, I don't know if it's actually considered better style to do it that way (I think that was added to make code generation easier), so defer to rustfmt if it tells you to not have the leading `|`.

Reply:
This was run through `rustfmt`, but I'm not sure it would change 

```x | y | z```

to 

```
| x
| y
| z
```

I can test for science :wink: